src/Tests/SmsFrameworkDeliveryReportTest.php

```diff
@@ -75,49 +68,88 @@ public function testDeliveryReports() {
   public function testDeliveryReportUpdate() {
```

Comment:
Lets not do these massive test web tests. The problem is we are testing a whole slew of things.

We can either test whether the automatic delivery report route is being created. Or we can **test processing a delivery report**.

Testing the delivery report URLs is its own test. This test should already exist. `SmsFrameworkDeliveryReportTest` is a legacy test. Lets not add on to it.

We shouldnt be dealing with HTTP/Json at all.

For example I created `\Drupal\Tests\sms\Functional\SmsFrameworkIncomingBrowserTest::testIncomingRouteEndpoint` to test the incoming route. The payload of the http request is irrelevant. It only tests whether the payload was received.

Instead this test should be a kernel test, directly passing an already created `SmsDeliveryReportEvent` object to the event dispatcher.